cpp/src/parquet/arrow/schema_internal.h

```diff
@@ -28,18 +29,19 @@ namespace parquet::arrow {
 
 using ::arrow::Result;
 
-Result<std::shared_ptr<::arrow::DataType>> FromByteArray(const LogicalType& logical_type);
+Result<std::shared_ptr<::arrow::DataType>> FromByteArray(const LogicalType& logical_type,
+                                                         bool use_known_arrow_extensions);
 Result<std::shared_ptr<::arrow::DataType>> FromFLBA(const LogicalType& logical_type,
                                                     int32_t physical_length);
 Result<std::shared_ptr<::arrow::DataType>> FromInt32(const LogicalType& logical_type);
 Result<std::shared_ptr<::arrow::DataType>> FromInt64(const LogicalType& logical_type);
 
 Result<std::shared_ptr<::arrow::DataType>> GetArrowType(
     Type::type physical_type, const LogicalType& logical_type, int type_length,
-    ::arrow::TimeUnit::type int96_arrow_time_unit = ::arrow::TimeUnit::NANO);
+    const ArrowReaderProperties& reader_properties);
```

Comment:
👍
Separate issue: @pitrou should we also introduce a subpart like `struct type_coerce`? I'm currently considering read large binary / binary view / dict[xx], currently the configurations would be a lot

Reply:
Do we need to do something about this here or is this a separate discussion? (I don't mind discussing it here if it's too small for a new issue, just checking if I should do something here).

Reply:
I don't really understand @mapleFU 's suggestion. It sounds unrelated to this PR.

Reply:
I've opened https://github.com/apache/arrow/issues/43810

Reply:
@pitrou 

> It sounds unrelated to this PR.

Maybe yes. Now that we have a lots of different properties to read "string" out. I think we need some better design on this...